WordPress/gutenberg
# [RNMobile] Search block: Add button position UI support

## Description
This PR builds on the branch for #29045 and as such, that PR must be merged before this one can be taken out of draft. This PR is a continuation in the Saga to port the Search Block to mobile. It adds the UI elements for the "button position" options:
- `button-inside`: search button is inside the borders of the search text input box. 
- `button-outside`: search button is outside the borders of the search text input box.
- `no-button`: search button is completely removed. 

This PR also removes the temporary alert messages used for testing in #29045, as well as cleans up some of the styling. 

Special note:
- The styles for this component are not yet finalized so the styling done here is just to get the overall layout of the button position UI implemented. The final version may have some variations. There will be another PR that focuses strictly on implementing styles and themes.

### Gutenberg Mobile PR
https://github.com/wordpress-mobile/gutenberg-mobile/pull/3196

## How has this been tested?



Tested inside the demo and WordPress apps for Android and iOS (emulator and simulator). Below are the testing instructions:

1. Add the search block to a post. The default button position setting will be `button-outside`. Verify the layout matches the default option.
2. Change the button position to **Button inside**. Verify button placement. 
3. Toggle the **icon only** option - verify **Button inside** is still valid. 
4. Change the button position to **Button outside**. Verify button placement. 
5. Change the button position option to **No button**. Verify the button disappears and the text input box expands to fill the horizontal space. 

## Screenshots 

### Button Outside

![button-pos-outside-text](https://user-images.githubusercontent.com/5810477/108278914-c7964f00-7149-11eb-89d0-c3c72ff25283.png)

![button-pos-outside-icon](https://user-images.githubusercontent.com/5810477/108278929-cc5b0300-7149-11eb-9323-2bee8a4d90a5.png)

### Button Inside

![button-pos-inside-text](https://user-images.githubusercontent.com/5810477/108278958-d7159800-7149-11eb-9502-b7d9b9a30df4.png)

![button-pos-inside-icon](https://user-images.githubusercontent.com/5810477/108278962-d977f200-7149-11eb-95bf-cc2d758a36b2.png)

### No Button

![no-button](https://user-images.githubusercontent.com/5810477/108278987-e09f0000-7149-11eb-9bfd-96528188d10f.png)


## Types of changes




This is a new feature for the mobile platform that will be spread across multiple PRs. The main issue for this project can be found [here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3073). 

## Checklist:
- [x] My code follows the accessibility standards.  **(will be handled in later PRs)**
- [x] I've tested my changes with keyboard and screen readers.  **(will be handled in later PRs)**
